compiler/rustc_hir_analysis/src/check/compare_impl_item.rs

```diff
@@ -2034,7 +2034,7 @@ pub(super) fn check_type_bounds<'tcx>(
         ObligationCause::new(impl_ty_span, impl_ty_def_id, code)
     };
 
-    let obligations = tcx
+    let obligations: Vec<_> = tcx
```

Comment:
We don't have to `collect` here anymore, do we?

Reply:
We don't *need* to, but there's a debug line right below.

Reply:
`compare_impl_item` isn't super perf sensitive anyways.